include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -189,68 +203,124 @@ struct BFloat16_t {
 
 static_assert(sizeof(BFloat16_t) == sizeof(uint16_t), "Sizes must match");
 
-/** \brief Used internally by the C++ API. C++ wrapper types inherit from this
+namespace detail {
+/** \brief This is a tagging template type. Use it with Base<T> to indicate that the C++ interface object
+ *   has no ownership of the underlying C object even though it is not being const.
+ */
+template <typename T>
+struct Unowned {
+  using Type = T;
+};
+
+/** \brief Used internally by the C++ API. C++ wrapper types inherit from this.
+ *   This is a zero cost abstraction to wrap the C API objects and delete them on destruction.
  *
- * This is a zero cost abstraction to wrap the C API objects and delete them on destruction.
- * There is a secondary class 'Unowned<T>' that is used to prevent deletion on destruction (Used for return types that are
- * not owned by the caller)
+ * All of the C++ classes function as
+ *  a) smart pointers that own objects that are created by the underlying C API. Their size is just
+ *  a pointer size, no need to dynamically allocate them. Use them by value.
+ *  b) convenient interfaces that return C++ objects and further enhance exception and type safety.
  *
  */
+
+/// <summary>
+/// This is a non-const pointer holder that is move-only. Disposes of the pointer on destruction.
+/// </summary>
 template <typename T>
 struct Base {
+ public:
```

Comment:
It's a struct, 'public' here is redundant?

Reply:
I agree, this is a leftover from my earlier experiments on how to solve the problems at hand.